### PR TITLE
Fix S3FileLoader local file clobbering

### LINKED ISSUE #735: S3FileLoader clears contents of local file when s3 object name == local file relative path name
### 🐛 Describe the bug

When an S3 object name matches the relative path of a local file, the file's contents get cleared after loading the object data.

```python
import torchdata
from torchdata.datapipes.iter import IterableWrapper, S3FileLoader


my_bucket = ...
local_file = ...
local_file_contents = "a,b,c,d\n1,2,3,4"

with open(local_file, "w") as outfile:
    outfile.write(local_file_contents)

with open(local_file, "r") as infile:
    file_text = infile.read()

print(file_text == local_file_contents)  # True

datapipe = IterableWrapper([f"s3://{my_bucket}/{local_file}"]).load_files_by_s3()

next(iter(datapipe))

with open(local_file, "r") as infile:
    file_text = infile.read()

print(file_text == local_file_contents)  # False
print(file_text == "")  # True
```

I am pretty sure it is bc of [this](https://github.com/pytorch/data/blob/main/torchdata/csrc/pybind/S3Handler/S3Handler.cpp#L205-L206). I think this could probably be fixed by opening the stream in a tempfile instead of just using `object_name_.c_str()`.

### Versions

```sh
$ python collect_env.py 
Collecting environment information...
PyTorch version: 1.12.1+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.9.12 (main, Apr  5 2022, 06:56:58)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-122-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: 11.5.119
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3090
Nvidia driver version: 495.29.05
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.0.5
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.0.5
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] facenet-pytorch==2.5.2
[pip3] mypy==0.971
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.0
[pip3] torch==1.12.1+cu113
[pip3] torchaudio==0.11.0+cu113
[pip3] torchdata==0.4.1
[pip3] torchvision==0.13.1+cu113
[conda] facenet-pytorch           2.5.2                    pypi_0    pypi
[conda] numpy                     1.22.0                   pypi_0    pypi
[conda] torch                     1.12.1+cu113             pypi_0    pypi
[conda] torchaudio                0.11.0+cu113             pypi_0    pypi
[conda] torchdata                 0.4.1                    pypi_0    pypi
[conda] torchvision               0.13.1+cu113             pypi_0    pypi
```

### PR DESCRIPTION
**Pull request by ringohoffman**
Fixes #735; remove unnecessary call to `storeFile()`, a remnant from the `aws-doc-sdk-examples` `transferOnStream.cpp` example, where this function is used to verify the buffer's contents: https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/cpp/example_code/transfer-manager/transferOnStream.cpp#L131

### Changes

- remove unnecessary call to `storeFile()`

### Testing

Verified files are no longer clobbered through local testing

### CODE REVIEW

**Review by NivekT — COMMENTED** [review]
Deferring to @ejguan and @ydaiming for review

**Comment by ringohoffman** [comment]
Based on the test failures, it seems like an extra file was added to the `s3://ai2-public-datasets/` bucket... shall I go ahead and update 77->78 and 39->40?

**Review by ejguan — APPROVED** [review]
Thank you. LGTM. I am fixing the tests in https://github.com/pytorch/data/pull/896. Will let you know when the fix has been landed and please rebase afterwards.

**Comment by ejguan** [comment]
> Based on the test failures, it seems like an extra file was added to the `s3://ai2-public-datasets/` bucket... shall I go ahead and update 77->78 and 39->40?

Yea. And, I personally think the test failures are unrelated to your PR. To guarantee the changes in this PR, let's wait until my PR is landed and rebase this PR to main branch afterwards. Thank you

**Comment by ejguan** [comment]
@ringohoffman Do you mind rebasing your commits onto main branch?

**Review by ejguan — APPROVED** [review]
LGTM. Thank you

**Pull request closed without merging**
